bpf/bpf_host.c

```diff
@@ -200,19 +201,26 @@ ipv6_host_policy_egress(struct __ctx_buff *ctx, __u32 srcID)
 	verdict = policy_can_egress6(ctx, &tuple, srcID, dstID,
 				     &policy_match_type);
 
+#ifdef POLICY_AUDIT_MODE
+	if (IS_ERR(verdict)) {
+		verdict = CTX_ACT_OK;
+		audited = 1;
+	}
+#endif
```

Comment:
We're hitting the BPF complexity limit on 4.19 (https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Kernel/1872/testReport/). You can think of the BPF complexity as the number of instructions on all paths through the program. (The verifier will in reality prune some of these paths and not walk them, but it's a good first approximation to work with complexity issues.)

Could you try moving the `if` block above inside `send_policy_verdict_policy()`. I suspect it might help a bit. If it's not enough, we might have to implement this bit of code in userspace, on the Go side.

Reply:
Removed unnecessary branching based on @joestringer suggestion (https://github.com/cilium/cilium/pull/11843#pullrequestreview-423209742)

Reply:
Thanks @ap4y , I've re-kicked the 4.19 CI so that will hopefully give us signal on whether this resolves the issue in the next 1-1.5 hours.